src/dbnode/persist/fs/commitlog/commit_log.go

```diff
@@ -512,7 +561,13 @@ func (l *commitLog) write() {
 		}
 		numDequeued = len(batch)
 
-		for _, writeBatch := range batch {
+		lastWritableElem := -1
+		for i := numDequeued - 1; i >= 0 && lastWritableElem == -1; i-- {
+			if batch[i].Err == nil && !batch[i].SkipWrite {
+				lastWritableElem = i
```

Comment:
nit: can this `break` instead of having `lastWritableElem == -1` in the guard? makes intentions clearer, and one fewer check per iteration is a bonus